docs/PIPELINE_DEPLOYMENTS.md

```diff
@@ -0,0 +1,169 @@
+- [Private Google APIs](https://cloud.google.com/vpc/docs/configure-private-google-access).
+- [Restricted gcr.io](https://cloud.google.com/vpc-service-controls/docs/set-up-gke#configure-dns).
+- [Restricted Artifact Registry](https://cloud.google.com/vpc-service-controls/docs/set-up-gke#configure-dns).
+
+## Best Practice for Dataflow Flex Template Usage
+
+The *Secured Data Warehouse Module* provides resources to deploy secured Dataflow Pipeline.
+We highly recommend you to use them to deploy your Dataflow Pipeline.
+
+### Temporary and Staging Location
+
+Use the appropriated [output](../README.md#outputs) of the main module as the Temporary and Staging Location bucket in the
+[pipeline options](https://cloud.google.com/dataflow/docs/guides/setting-pipeline-options#setting_required_options):
+
+- Data ingestion project: `data_ingestion_dataflow_bucket_name`.
+- Confidential Data project: `confidential_data_dataflow_bucket_name`.
+
+### Dataflow Worker Service Account
+
+Use the appropriated [output](../README.md#outputs) of the main module as the
+[Dataflow Controller Service Account](https://cloud.google.com/dataflow/docs/concepts/security-and-permissions#specifying_a_user-managed_worker_service_account):
+
+- Data ingestion project: `dataflow_controller_service_account_email`.
+- Confidential Data project: `confidential_dataflow_controller_service_account_email`.
+
+### Customer Managed Encryption Key
+
+Use the appropriated [output](../README.md#outputs) of the main module as the [Dataflow KMS Key](https://cloud.google.com/dataflow/docs/guides/customer-managed-encryption-keys):
+
+- Data ingestion project: `cmek_data_ingestion_crypto_key`
+- Confidential project: `cmek_reidentification_crypto_key`
+
+### Disable Public IPs
+
+[Disabling Public IPs helps to better secure you data processing infrastructure.](https://cloud.google.com/dataflow/docs/guides/routes-firewall#turn_off_external_ip_address).
+Make sure you have your subnetwork configured as [Subnetwork section](#subnetwork) details.
+
+### Enable Streaming Engine
+
+Enabling Streaming Engine it is important to ensure all the performance benefits of the infrastructure. You can learn more about it in the [documentation](https://cloud.google.com/dataflow/docs/guides/deploying-a-pipeline#streaming-engine).
+
+## Deploying Dataflow Flex Jobs
+
+We recommend the usage of [Flex Job Templates](https://cloud.google.com/dataflow/docs/guides/templates/using-flex-templates).
+You can learn more about the differences between Classic and Flex Templates [here](https://cloud.google.com/dataflow/docs/concepts/dataflow-templates#evaluating-which-template-type-to-use).
+
+### Deploying with Terraform
+
+Use the Dataflow Flex Job Template [submodule](../modules/dataflow-flex-job/README.md).
+
+### Deploying with `gcloud` Command
+
+You can run the following commands to create a **Java** Dataflow Flex Job using the **gcloud command**:
+
```

Comment:
you might mention that these variables are from the outputs of the main module as shared in the earlier section.